### PR TITLE
[F2F-836 and F2F-823]:Update postEvent to Save clientLandingPageUrl and remove PII from postEvent

### PR DESCRIPTION
**Pull request by richa-misra**
### Issue tracking

- [F2F-836](https://govukverify.atlassian.net/browse/F2F-836)
- [F2F-823](https://govukverify.atlassian.net/browse/F2F-823)

### PII logging

- [X] Verified that no PII data is being logged

### Testing

**AUTH_IPV_REQUESTED**

Screenshot 2023-06-23 at 15 00 24
Screenshot 2023-06-23 at 15 00 37

**YOTI_START**
Screenshot 2023-06-23 at 15 32 10

**VC_CONSUMED**
Screenshot 2023-06-23 at 15 33 04

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[F2F-836]: https://govukverify.atlassian.net/browse/F2F-836?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[F2F-823]: https://govukverify.atlassian.net/browse/F2F-823?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "target": "es2020",
     "strict": true,
     "preserveConstEnums": true,
-    "sourceMap": false,
+    "sourceMap": true,
```
Comment (darrendunford) [inline src/tsconfig.json:7]: nit: this shouldn't be merged to main

File: src/PostEventHandler.ts [hunk 2]
```diff
@@ -20,9 +20,15 @@ class PostEventHandler implements LambdaInterface {
...
 	async handler(event: SQSEvent, context: any): Promise<SQSBatchResponse> {
 		if (event.Records.length === 1) {
+			let body;
 			const record: SQSRecord = event.Records[0];
 
-			const body = JSON.parse(record.body);
+			try {
+				body = JSON.parse(record.body);
+			} catch (exp) {
+				logger.error({ message:"Received invalid JSON in the SQS event record.body" });
```
[the remark this answers is not in the document]
Comment (richa-misra) [reply src/PostEventHandler.ts:29]: Don't see extra benefit of logging the actual exception as if its a JSON.parse, would mean its an invalid json  
Screenshot 2023-06-26 at 11 36 44

File: src/models/ReturnSQSEvent.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ export type EventType = "AUTH_IPV_AUTHORISATION_REQUESTED" | "F2F_YOTI_START" |
 export interface ReturnSQSEvent {
 	event_id: string;
 	client_id: string;
-	component_id: string;
+	clientLandingPageUrl: string;
```
Comment (darrendunford) [inline src/models/ReturnSQSEvent.ts:8]: Even though we don't use it will we still get component_id (optionally) in our event payloads?

**Comment by darrendunford** [comment]
the ticket in the branch and the PR title don't match?

File: src/models/ReturnSQSEvent.ts (hunk 3 above)
Comment (richa-misra) [reply src/models/ReturnSQSEvent.ts:8]: we won't receive `component_id` as IPR not using that, the existing fields/events that IPR have asked for are configured here: https://github.com/alphagov/di-txma-event-self-serve/blob/main/config/consumer/sqs/return.json

File: src/PostEventHandler.ts [hunk 4]
```diff
@@ -19,10 +19,21 @@ class PostEventHandler implements LambdaInterface {
...
+
 		if (event.Records.length === 1) {
+			let body;
 			const record: SQSRecord = event.Records[0];
 
-			const body = JSON.parse(record.body);
+			try {
+				body = JSON.parse(record.body);
+			} catch (exp) {
+				logger.error({ message:"Received invalid JSON in the SQS event record.body" });
```
Comment (darrendunford) [inline src/PostEventHandler.ts:34]: how about:
```suggestion
			} catch (error) {
				logger.error({ message:"Received invalid JSON in the SQS event record.body", error });
```
?

**Review by darrendunford — DISMISSED** [review]

**Pull request merged**